nrf_security/src/mbedtls/shared/CMakeLists.txt

```
@@ -36,25 +20,54 @@ add_library(mbedcrypto_shared STATIC
   ${ZEPHYR_BASE}/misc/empty_file.c
 )
 
-target_sources_ifdef(CONFIG_VANILLA_MBEDTLS_CHACHA20_C
-  mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/chacha20.c
-)
-target_sources_ifdef(CONFIG_VANILLA_MBEDTLS_POLY1305_C
-  mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/poly1305.c
-)
-target_sources_ifdef(CONFIG_VANILLA_MBEDTLS_CHACHAPOLY_C
-  mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/chachapoly.c
-)
-target_sources_ifdef(CONFIG_VANILLA_MBEDTLS_RSA_C
-  mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/rsa.c
-)
-target_sources_ifdef(CONFIG_VANILLA_MBEDTLS_RSA_C
-  mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/rsa_internal.c
-)
+if (CONFIG_VANILLA_MBEDTLS_CHACHA20_C OR
+  (CONFIG_MBEDTLS_CHACHA20_C AND NOT CONFIG_CC310_MBEDTLS_CHACHA20_C))
+
+  target_sources(mbedcrypto_shared PRIVATE
+    ${ARM_MBEDTLS_PATH}/library/chacha20.c
+  )
+endif()
+
+if (CONFIG_VANILLA_MBEDTLS_POLY1305_C OR
+  (CONFIG_MBEDTLS_POLY1305_C AND NOT CONFIG_CC310_MBEDTLS_POLY1305_C))
+
+  target_sources(mbedcrypto_shared PRIVATE
+    ${ARM_MBEDTLS_PATH}/library/poly1305.c
+  )
+endif()
+
+if (CONFIG_VANILLA_MBEDTLS_CHACHAPOLY_C OR
+  (CONFIG_MBEDTLS_CHACHAPOLY_C AND NOT CONFIG_CC310_MBEDTLS_CHACHAPOLY_C))
+
+  target_sources(mbedcrypto_shared PRIVATE
+    ${ARM_MBEDTLS_PATH}/library/chachapoly.c
+  )
+endif()
+
+if (CONFIG_VANILLA_MBEDTLS_RSA_C OR
+   (CONFIG_MBEDTLS_RSA_C AND NOT CONFIG_CC310_MBEDTLS_RSA_C))
+
+  target_sources(mbedcrypto_shared PRIVATE
+    ${ARM_MBEDTLS_PATH}/library/rsa.c
+    ${ARM_MBEDTLS_PATH}/library/rsa_internal.c
+  )
+
+endif()
+
 target_sources_ifdef(CONFIG_MBEDTLS_SHA512_C
   mbedcrypto_shared PRIVATE ${ARM_MBEDTLS_PATH}/library/sha512.c
 )
 
+# Special case: Oberon only backend and enabled DHM
+if (CONFIG_MBEDTLS_DHM_C AND NOT NRF_CRYPTO_BACKEND_COMBINATION_0 AND NOT CONFIG_CC310_MBEDTLS_DHM_C)
```

Based on your comment, wouldn't this be a little more correct ?
```suggestion
if (CONFIG_MBEDTLS_DHM_C AND CONFIG_OBERON_BACKEND)
```

after all, `CONFIG_CC310_MBEDTLS_DHM_C` can only be true if `NRF_CRYPTO_BACKEND_COMBINATION_0` is also true.

https://github.com/nrfconnect/sdk-nrfxlib/blob/master/nrf_security/Kconfig#L682-L711

Secondly, `CONFIG_MBEDTLS_DHM_C` cannot be true, if `NRF_CRYPTO_BACKEND_COMBINATION_0` is also true:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/nrf_security/Kconfig#L671-L680
specifically because of:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/nrf_security/Kconfig#L675

---

Changed to CONFIG_MBEDTLS_DHM_C AND CONFIG_OBERON_BACKEND instead